CS361_Project/tests/unitTests/test_Login.py

```diff
@@ -0,0 +1,44 @@
+from django.test import TestCase
+from unittest.mock import Mock
+from CS361_Project.functions import *
+from CS361_Project.models import Account
+
+
+class TestLogin(TestCase):
+    def setUp(self):
+        self.mock_user = Mock()
+        self.mock_user.username = 'username'
+        self.mock_user.password = 'password'
+        self.mock_user.name = 'name'
+        self.mock_user.role = 1
+        
+    def test_Login(self):
+        mock_request = Mock()
+        mock_request.POST = {
+            "username": "username",
+            "password": "password",
+            'LoggedIn' : False
+        }
+        Management.User.login(mock_request, self.mock_user)
+        
+        # Assert that session variables are set correctly
+        self.assertEqual(mock_request.session['name'], self.mock_user.name)
+        self.assertEqual(mock_request.session['role'], self.mock_user.role)
+        self.assertTrue(mock_request.session['LoggedIn'])
+    def test_authenticate_user(self):
+         mock_user = Mock()
+         mock_user.username = "test_user"
+         mock_user.password = "test_password"
+         #TODO: not sure if this is even correct, any other way to mock an account that will show in get?
+         Account.objects.get = Mock(return_value=mock_user)
```

Comment:
Can someone check this? Im not sure if this is correct

Reply:
just resolved the issue, turns out Account.objects.get = Mock(return_value=mock_user) works just fine. All the login unit tests are passing and the other unit tests are mostly passing, ill try to finish the rest of the manageAccount tests today.